cdist/exec/local.py

```diff
@@ -224,8 +224,16 @@ def save_cache(self):
                 shutil.rmtree(destination)
         except PermissionError as e:
             raise cdist.Error("Cannot delete old cache %s: %s" % (destination, e))
-
-        shutil.move(self.base_path, destination)
+        # By default, if available, cdist uses ssh/scp with
+        # multiplexing options. cdist uses user specified out_path
+        # or temp directory for storing files and in the end it
+        # saves this cache to cache directory.
+        # It is an error to move/copy mux ControlPath socket file
+        # to cache so it needs to be ignored when saving to cache.
+        shutil.copytree(self.base_path, destination,
+                ignore=shutil.ignore_patterns(
+                    '*' + cdist.MUX_SOCKET_PREFIX + '*'))
+        shutil.rmtree(self.base_path)
```

Comment:
Python 3.5 introduced copy_function argument to shutil.move that could be used for this functionality, but Python 3.2 that is minimal python version does not have it so I am using copytree and rmtree.